js/ui/file_manager/file_items_controller.js

```diff
@@ -147,50 +147,81 @@ export default class FileItemsController {
     }
 
     createDirectory(parentDirectoryInfo, name) {
-        return when(this._fileProvider.createFolder(parentDirectoryInfo.fileItem, name))
-            .done(() => this._resetDirectoryState(parentDirectoryInfo));
+        return this._fileProvider.createFolder(parentDirectoryInfo.fileItem, name);
+    }
+
+    completeCreateDirectory(parentDirectoryInfo) {
```

Comment:
It is a bad approach when you call action, that should always call from another entity. It is potentional place for a big bugs in future. I think, the best solution is revert the complete... changes, and solve the problem in future PR.

In object that call complete... methods you can define empty promise.

Reply:
We can't revert the complete... changes without big refactoring. The previous version of the code already has many bugs that are fixed in the current version of the code. So, I think we should do this kind of refactoring in the future PR.